### PR TITLE
add alert component

### PR DESCRIPTION
**Pull request by ajronn**

### CODE REVIEW

File: src/core/alert/AlertProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+const Context = createContext(STATE);
+
+export const AlertProvider = (props: any) => {
+    const [alerts, setAlerts] = useState<Alert[]>([]);
+
+    // const sleep = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+    const addAlert = (msg: string) => {
+        const id = alerts.length > 0 ? alerts[alerts.length - 1].id + 1 : 0;
```
Comment (Czaroos) [inline src/core/alert/AlertProvider.tsx:32]: We don't need explicit id when we use array you can remove it and simplify it as mentioned below

File: src/core/alert/AlertProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    const state: Props = {
+        alerts: alerts,
+        addAlert: addAlert
+    }
+
+    return (
+        <Context.Provider value={state}>
+            {alerts.length > 0 &&
+                <AlertsContainer>
+                    {alerts.map((e) => {
```
Comment (Czaroos) [inline src/core/alert/AlertProvider.tsx:62]: change e to something meaningful (e indicates event, whereas it's alert obj) and add second arg index provided with .map function

File: src/core/alert/AlertProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                        return <Alert key={e.id} msg={e.msg + e.id} onClick={() => removeAlert(e.id)} variant={e.variant} />
```
Comment (Czaroos) [inline src/core/alert/AlertProvider.tsx]: change .id to index

File: src/core/alert/AlertProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const [alerts, setAlerts] = useState<Alert[]>([]);
+
+    // const sleep = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+    const addAlert = (msg: string) => {
+        const id = alerts.length > 0 ? alerts[alerts.length - 1].id + 1 : 0;
+        setAlerts([...alerts, { msg: msg, id: id, variant: false }]);
+    }
+
+    const removeAlert = async (id: number) => {
```
Comment (Czaroos) [inline src/core/alert/AlertProvider.tsx:36]: index as arg and simply remove this element from array

File: src/core/alert/style.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import styled from "styled-components";
+
+export const AlertsContainer = styled.div`
+    width: 500px;
+    height: auto;
```
Comment (Czaroos) [inline src/core/alert/style.ts]: its implicitly  auto as default so you can remove this line

**Review by Czaroos — CHANGES_REQUESTED** [review]

**Review by i0skar — APPROVED** [review]

File: src/core/alert/AlertProvider.tsx (hunk 2 above)
Comment (ajronn) [reply src/core/alert/AlertProvider.tsx:62]: In map function e is not an event, this is element... e mean event in section of call function, not in arguments section

File: src/core/alert/AlertProvider.tsx (hunk 1 above)
Comment (ajronn) [reply src/core/alert/AlertProvider.tsx:32]: Id is necessary, cause remove fun is async, if we will be based on an index that changes, we will remove wrong alert... 
the same for animations

File: src/core/alert/AlertProvider.tsx (hunk 3 above)
Comment (ajronn) [reply src/core/alert/AlertProvider.tsx]: Index changes, no sense

File: src/core/alert/AlertProvider.tsx (hunk 4 above)
Comment (ajronn) [reply src/core/alert/AlertProvider.tsx:36]: same as above

File: src/core/alert/style.ts (hunk 5 above)
Comment (ajronn) [reply src/core/alert/style.ts]: done

**Review by Czaroos — APPROVED** [review]

**Pull request merged**